modules/core/src/test/java/org/apache/ignite/internal/processors/database/BPlusTreeSelfTest.java

```diff
@@ -208,10 +208,22 @@ protected ReuseList createReuseList(int cacheId, PageMemory pageMem, long rootId
 
                 try {
                     asyncRunFut.cancel();
-                    asyncRunFut.get(60000);
+                    asyncRunFut.get(TimeUnit.MINUTES.toMillis(1));
                 }
                 catch (Throwable ex) {
-                    //Ignore
+                    // Ignore.
+                }
+
+                // We should try to wait for all the futures to complete, since GridCompoundFuture#get(long) fails if
```

Comment:
I don't understand this comment: how does waiting for all futures prevent JVM crash?

Reply:
oh, looks like I get it: the above compound future can fail, while other futures are still running

Reply:
Yep